ISBN/src/c/ISBN.php

```diff
@@ -16,6 +16,7 @@ public function __construct(string $isbn)
     private function ensureValid(string $isbn)
     {
         $this->ensureRightPrefix($isbn);
+        $this->ensureRightNumberOfGroups($isbn);
```

Comment:
ich würde $this->splitIsbn() hier aufrufen und danach das array als parameter für ensureRightNumberOfGroups() und ensureRightGroupNumber() verwenden. so muss die isbn nur 1x durch den verhältnismässig teuren regex aufgesplittet werden.